frame/grandpa/src/lib.rs

```diff
@@ -121,6 +121,15 @@ pub mod pallet {
 		/// Max Authorities in use
 		#[pallet::constant]
 		type MaxAuthorities: Get<u32>;
+
+		/// The maximum number of entries to keep in the set id to sesion index mapping.
+		///
+		/// Since the `SetIdSession` map is only used for validating equivocations this
+		/// value should relate to the bonding duration of whatever staking system is
+		/// being used (if any). If equivocation handling is not enabled then this value
+		/// can be zero.
+		#[pallet::constant]
+		type MaxSetIdSessionEntries: Get<u32>;
```

Comment:
This is due to a mismatch between set id type (`u64`) and session index type (`u32`). If I change the type we can remove some of the casts below, but instead will have to introduce casts when defining this constant in the runtime definition (since all the staking and session durations are defined in `u32`). I'd rather keep it as is.

Reply:
Thanks to this smart guy 🫡 updated